src/team_lib/mhba/xccl_mhba_collective.c

```diff
@@ -39,7 +38,7 @@ static xccl_status_t xccl_mhba_node_fanin(xccl_mhba_team_t     *team,
                                    request); // no option for failure status
 
     if (team->node.sbgp->group_rank != team->node.asr_rank) {
-        *team->node.ops[index].my_ctrl = request->seq_num;
+        *team->node.ops[index].my_ctrl = request->self_change_flag ? -request->seq_num : request->seq_num;
```

Comment:
in 0 we'll always have to update, so line 59 takes care of it. because for asr self_change flag will always be 1 for 0. ill add a comment explainig this

Reply:
after the fix, seq_num 0 won't matter because msg_size will be different, as it set to 0 in the beginning